recipes/mysql-server.rb

```diff
@@ -35,11 +35,13 @@
         action [:create, :start]
     end
 
-    mysql_config 'default' do
-        source 'mysql_config.erb'
-        notifies :restart, 'mysql_service[default]'
-        action :create
-    end
+    if node['mysql']['server_version'] != "5.5"
```

Comment:
Those tweaks are important for Magento regardless of MySQL version.  e.g. without max_allowed_packet you probably won't be able to import most Magento database dumps.

I'm not sure there's any version of MySQL that supports resizing of log files (unless that's happened relatively recently), and for dev/ci instances (which is the primary use case here) removing the log files should be a pretty safe thing to do.  So I'd go with @aligenttimaslat 's plan of stop the service, delete the files and start it again after the config change, but for all MySQL versions not just 5.5.